processor/src/host/mod.rs

```diff
@@ -111,16 +111,17 @@ pub trait AsyncHost: BaseHost {
 
     /// Returns MAST forest corresponding to the specified digest, or None if the MAST forest for
     /// this digest could not be found in this [AsyncHost].
-    fn get_mast_forest(
-        &self,
-        node_digest: &Word,
-    ) -> impl Future<Output = Option<Arc<MastForest>>> + Send;
+    fn get_mast_forest(&self, node_digest: &Word) -> impl SFuture<Option<Arc<MastForest>>>;
 
     /// Handles the event emitted from the VM and provides advice mutations to be applied to
     /// the advice provider.
     fn on_event(
         &mut self,
         process: &ProcessState<'_>,
         event_id: u32,
-    ) -> impl Future<Output = Result<Vec<AdviceMutation>, EventError>> + Send;
+    ) -> impl SFuture<Result<Vec<AdviceMutation>, EventError>>;
 }
+
+pub trait SFuture<O>: Future<Output = O> + Send {}
```

Comment:
If we _really_ need a type alias like this (and I would argue we don't), then we should call it `SendFuture` so it is at least immediately obvious what it is. I hadn't the slightest idea from just `SFuture` alone.

Reply:
I would remove this trait - `impl Future<...> + Send` is fine and used everywhere in the ecosystem

Reply:
Assuming the approach holds, the return is required, otherwise we'd have `#[cfg(..)]` not only in place of the trait declaration, but also the implementations.

Reply:
Good point - my assumption was that downstream implementations were going to need to choose whether they target wasm, or everything else. But maybe that's the wrong assumption to make